core/base/single_thread_dispatcher.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <deque>
+#include <memory>
+#include <mutex>
+#include <functional>
+#include <atomic>
+
+#include "message_window.h"
+namespace foxglove {
+namespace windows {
+/**
+ * SingleThreadDispatcher
+ * All task enqueued will be run on the thread where the dispatcherr is created.
+ */
+
+class SingleThreadDispatcher {
+ public:
+  using Task = std::function<void()>;
+
+  SingleThreadDispatcher();
+  ~SingleThreadDispatcher();
+
+  void Dispatch(Task task);
+  void Terminate();
+ private:
+  void ProcessTasks();
+  std::vector<Task> tasks_;
```

Comment:
nit: `std::dequeue` may be more suitable for storing tasks as it won’t reallocate upon appending items

Reply:
moved to dequeue